### ISSUE TITLE
Empty enum arguments rule matches method calls

### ISSUE THREAD

**Issue by colinta** [realm/SwiftLint#1625]
Example:

```
switch fontName {  // a String
case UIFont.editorItalicFont().fontName:  // we want to see if it matches this font name
...  // we get a warning on ^ that line "Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated..."
```

**Comment by marcelofabri** [comment]
This should be fixed on master, can you try running with it?

**Comment by colinta** [comment]
I built SwiftLint from source and got the same warnings. 🙁

**Comment by marcelofabri** [comment]
I added it as an example in `nonTriggeringExamples` and it didn't trigger 🤔 

Are you sure you're running the version you've built from `master`? 😅

**Comment by colinta** [comment]
I removed the version installed by homebrew just to be sure, then pulled master again, and ran `sudo make install` one more time.  Is there a way to check the version?  (oddly, `-v / --version / -h / --help` do nothing, that's unexpected)

The repo I'm running this against is public: https://github.com/ello/ello-ios

**Comment by marcelofabri** [comment]
`swiftlint version` is the command, but for master it'd return the same thing as the latest version 😅 

I'll take a look at this later.

**Comment by marcelofabri** [comment]
These are all `empty_enum_arguments ` violations I've got:

```
/Users/marcelofabri/dev/ello-ios/Sources/Controllers/Omnibar/OmnibarScreenTableViewDelegate.swift:64:33: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated types if they are not used. (empty_enum_arguments)
/Users/marcelofabri/dev/ello-ios/Sources/Networking/ElloAPI.swift:351:32: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated types if they are not used. (empty_enum_arguments)
/Users/marcelofabri/dev/ello-ios/Sources/Networking/ElloAPI.swift:380:35: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated types if they are not used. (empty_enum_arguments)
/Users/marcelofabri/dev/ello-ios/Sources/Networking/ElloAPI.swift:416:32: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated types if they are not used. (empty_enum_arguments)
/Users/marcelofabri/dev/ello-ios/Sources/Networking/ElloAPI.swift:436:35: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated types if they are not used. (empty_enum_arguments)
```

They're seem valid to me.

**Comment by colinta** [comment]
I don't know what happened, but my previous attempts at compiling must not have worked, after changing the version to make sure I was running from source I just reran it against my code and, low and behold, no warnings! (I had already fixed the ones you found in my local branch).

Thanks for bearing with me, I can verify it's working in master.  Sorry it took so long to get that working!

**Comment by colinta** [comment]
(I had this comment lined up in another tab, never hit submit)

I modified the source to change the version to `0.19.0-master` and recompiled, and `swiftlint version` outputs that version; so I'm definitely running the compiled-from-source binary.

**Comment by marcelofabri** [comment]
No worries, glad we could sort it out 💯
